src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java

```diff
@@ -0,0 +1,377 @@
+            new SameDayTestCase("1:23", _2020_08_13__02_59_59_EDT, false, true),
+
+            // Not same-day departures
+            new SameDayTestCase(QUERY_TIME, _2020_08_12__23_59_59_EDT, false, false),
+            new SameDayTestCase(QUERY_TIME, _2020_08_13__02_59_59_EDT, false, false),
+            new SameDayTestCase(QUERY_TIME, _2020_08_14__03_00_00_EDT, false, false),
+            new SameDayTestCase("1:23", _2020_08_13__03_00_00_EDT, false, false),
+            new SameDayTestCase("1:23", _2020_08_13__23_59_59_EDT, false, false),
+            new SameDayTestCase("1:23", _2020_08_14__02_59_59_EDT, false, false),
+
+            // Same-day arrivals
+            new SameDayTestCase(QUERY_TIME, _2020_08_13__03_00_00_EDT, true, true),
+            new SameDayTestCase(QUERY_TIME, _2020_08_13__23_59_59_EDT, true, true),
+            new SameDayTestCase(QUERY_TIME, _2020_08_14__02_59_59_EDT, true, true),
+            new SameDayTestCase("1:23", _2020_08_12__03_00_00_EDT, true, true),
+            new SameDayTestCase("1:23", _2020_08_12__23_59_59_EDT, true, true),
+            new SameDayTestCase("1:23", _2020_08_13__02_59_59_EDT, true, true),
+
+            // Not same-day arrivals
+            new SameDayTestCase(QUERY_TIME, _2020_08_12__23_59_59_EDT, true, false),
+            new SameDayTestCase(QUERY_TIME, _2020_08_13__02_59_59_EDT, true, false),
+            new SameDayTestCase(QUERY_TIME, _2020_08_14__03_00_00_EDT, true, false),
+            new SameDayTestCase("1:23", _2020_08_13__03_00_00_EDT, true, false),
+            new SameDayTestCase("1:23", _2020_08_13__23_59_59_EDT, true, false),
+            new SameDayTestCase("1:23", _2020_08_14__02_59_59_EDT, true, false)
+        );
+    }
+
+    /**
+     * Check that only same-day itineraries are selected.
+     */
+    @Test
+    public void testGetSameDayItineraries() throws URISyntaxException {
+        MonitoredTrip trip = makeTestTrip(false);
+        trip.tripTime = QUERY_TIME;
+
+        List<Long> startTimes = List.of(
+            _2020_08_13__23_59_59_EDT, // same day
+            _2020_08_14__02_59_59_EDT, // considered same day
+            _2020_08_14__03_00_00_EDT // not same day
+        );
+
+        // Create itineraries, some being same-day, some not per the times above.
+        List<Itinerary> itineraries = startTimes.stream()
+            .map(t -> simpleItinerary(t, false))
+            .collect(Collectors.toList());
+
+        List<Itinerary> processedItineraries = ItineraryUtils.getSameDayItineraries(itineraries, trip, QUERY_DATE);
+        Assertions.assertEquals(2, processedItineraries.size());
+        Assertions.assertTrue(processedItineraries.contains(itineraries.get(0)));
+        Assertions.assertTrue(processedItineraries.contains(itineraries.get(1)));
+    }
+
+    /**
+     * Helper method to create a bare-bones itinerary with start or end time.
+     */
+    private Itinerary simpleItinerary(Long startTime, boolean isArrival) {
```

Comment:
Please rename `startTime` to `targetEpochMillis`.

Reply:
This was addressed in https://github.com/ibi-group/otp-middleware/pull/77/commits/78e31e7b50bcda047f65ddf1cbbbe506dbb8e455#diff-5a730e4259529c8d307574db916d01f209537dd2f60746a86a6e9b022ce8cca4R286.